crates/core/src/db/relational_db.rs

```diff
@@ -214,13 +215,28 @@ impl RelationalDB {
     }
 
     #[tracing::instrument(skip_all)]
-    pub fn schema_for_column(&self, tx: &MutTxId, table_id: u32, col_id: u32) -> Result<AlgebraicType, DBError> {
-        let schema = self.row_schema_for_table(tx, table_id)?;
-        let col = schema
-            .elements
-            .get(col_id as usize)
-            .ok_or(TableError::ColumnNotFound(col_id))?;
-        Ok(col.algebraic_type.clone())
+    pub fn schema_for_column<'tx>(
+        &self,
+        tx: &'tx MutTxId,
+        table_id: u32,
+        col_id: u32,
+    ) -> Result<Cow<'tx, AlgebraicType>, DBError> {
+        Ok(match self.row_schema_for_table(tx, table_id)? {
+            Cow::Borrowed(schema) => Cow::Borrowed(
+                &schema
+                    .elements
+                    .get(col_id as usize)
+                    .ok_or(TableError::ColumnNotFound(col_id))?
+                    .algebraic_type,
+            ),
+            Cow::Owned(mut schema) => {
```

Comment:
Mostly echoing what I have perceived to be Tyler's feedback around this part of the codebase: I would prefer comments aimed at a person who is not expert at Rust. Right here, I would take that to mean someone who is unfamiliar with the `Cow` interface, for instance. It's unclear why this path requires a manual bounds check!